### ISSUE TITLE
Unexpected escape analysis behavior for interpolated string with variable of local scope

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#63262]
The escape analysis behavior for the following interpolated string cases is unexpected:
```csharp
using System;
using System.Runtime.CompilerServices;

[InterpolatedStringHandler]
ref struct CustomHandler
{
    public CustomHandler(int literalLength, int formattedCount) { }
    public void AppendFormatted(Span<char> s) { }
}

class Program
{
    static CustomHandler F1()
    {
        Span<char> s = stackalloc char[10];
        CustomHandler h1 = $"{s}";
        return h1; // no error
    }

    static void F2()
    {
        Span<char> s = stackalloc char[10];
        F3($"{s}"); // error CS8352: ... may expose variables outside declaration scope
    }

    static void F3(ref CustomHandler handler) { }
}
```

For `F1()`, no error is reported returning `h1` although it may capture the local `Span<char>` from `$"{s}"`.

For `F2()`, an error is reported calling `F3($"{s}")`, although it seems no references should escape `F3()`.

**Comment by Youssef1313** [comment]
I both cases, can't it escape to a badly written `InterpolatedStringHandler`? So even a `_ = $"{s}";` should be an error? Anything I'm missing?

**Comment by cston** [comment]
The local span may be captured by `CustomHandler` instance, which is why `F1()` should report an error for `return h1;`. In `F2()` though, I don't believe the `CustomHandler` instance (or the `ref` to the instance) can escape `F3()` other than in the same `CustomHandler` instance.
